components/camel-platform-http-main/src/main/java/org/apache/camel/component/platform/http/main/MainHttpServer.java

```diff
@@ -465,17 +479,22 @@ protected void setupJolokia() {
         platformHttpComponent.addHttpEndpoint("/q/jolokia", null, null);
     }
 
+    protected PlatformHttpPluginRegistry resolvePlatformHttpPluginRegistry() {
+        Optional<PlatformHttpPluginRegistry> result = ResolverHelper.resolveService(
+                getCamelContext(),
+                PlatformHttpPluginRegistry.FACTORY,
+                PlatformHttpPluginRegistry.class);
+        return result.orElse(null);
+    }
+
     private HttpRequestHandler getHttpRequestHandler() {
-        //TODO: make jolokiaService more pluggable
-        //JolokiaHttpRequestHandlerSupport jolokiaService = camelContext.getCamelContextExtension().getContextPlugin(JolokiaHttpRequestHandlerSupport.class);
-        HttpRequestHandler requestHandler;
-        try (JolokiaHttpRequestHandlerSupport jolokiaService = new JolokiaHttpRequestHandlerSupport()) {
-            jolokiaService.start();
-            requestHandler = jolokiaService.getHttpRequestHandler();
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-        return requestHandler;
+        PlatformHttpPluginRegistry registry
+                = camelContext.getCamelContextExtension().getContextPlugin(PlatformHttpPluginRegistry.class);
+        JolokiaPlatformHttpPlugin jolokia
```

Comment:
Jolokia should only be discovered if jolokiaEnabled

Reply:
this is only called inside 
if (jolokiaEnabled) {
    setupJolokia();
}

Reply:
ok